src/main.rs

```diff
@@ -222,6 +226,257 @@ fn main() {
+
+            let mut txids_at_height = vec![];
+
+            for mblock in microblocks.iter() {
+                for tx in mblock.txs.iter() {
+                    tx_fees.insert(tx.txid(), tx.get_tx_fee());
+                    txids_at_height.push(tx.txid());
+                }
+            }
+
+            for tx in block.txs.iter() {
+                if tx.get_tx_fee() > 0 {
+                    // not a coinbase
+                    tx_fees.insert(tx.txid(), tx.get_tx_fee());
+                    txids_at_height.push(tx.txid());
+                }
+            }
+
+            tx_mined_heights.insert(block_info.height, txids_at_height);
+
+            // next block
+            block_info = match StacksChainState::load_staging_block_info(
+                chainstate.db(),
+                &StacksBlockHeader::make_index_block_hash(
+                    &block_info.parent_consensus_hash,
+                    &block_info.parent_anchored_block_hash,
+                ),
+            )
+            .unwrap()
+            {
+                Some(blk) => blk,
+                None => {
+                    break;
+                }
+            };
+            block_info.block_data = StacksChainState::load_block_bytes(
+                &chainstate.blocks_path,
+                &block_info.consensus_hash,
+                &block_info.anchored_block_hash,
+            )
+            .unwrap()
+            .expect("No such block");
+        }
+
+        let estimator = Box::new(UnitEstimator);
+        let metric = Box::new(UnitMetric);
+        let mempool_db =
+            MemPoolDB::open(true, CHAIN_ID_MAINNET, &chain_state_path, estimator, metric)
+                .expect("Failed to open mempool db");
+
+        let mut total_txs = 0;
+        for (_, txids) in tx_mined_heights.iter() {
+            total_txs += txids.len();
+        }
+
+        let mut tx_cnt = 0;
+        for (mined_height, txids) in tx_mined_heights.iter() {
+            for txid in txids.iter() {
+                tx_cnt += 1;
+                if tx_cnt % 100 == 0 {
+                    debug!("Check tx {} of {}", tx_cnt, total_txs);
+                }
+
+                if let Some(txinfo) = MemPoolDB::get_tx(&mempool_db.db, txid).unwrap() {
+                    let delta = mined_height.saturating_sub(txinfo.metadata.block_height);
+                    if let Some(txids_at_delta) = tx_mined_deltas.get_mut(&delta) {
+                        txids_at_delta.push(txid.clone());
+                    } else {
+                        tx_mined_deltas.insert(delta, vec![txid.clone()]);
+                    }
+                }
+            }
+        }
+
+        let mut deltas: Vec<_> = tx_mined_deltas.keys().collect();
+        deltas.sort();
+
+        let mut reports = vec![];
+        for delta in deltas {
+            let mut delta_tx_fees = vec![];
+            let empty_txids = vec![];
+            let txids = tx_mined_deltas.get(&delta).unwrap_or(&empty_txids);
```

Comment:
if there are no txids, maybe you should just `continue`. 

otherwise you might run into problems in computing stats. ex, `let median_tx_fee = delta_tx_fees[delta_tx_fees.len() / 2]` would panic